src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_CONSTANT = 1.5;
+    private static final int EMPTY_SIZE = 0;
+    private Object[] elementData;
```

Comment:
It is better to use T[] elementData here to avoid redundant casts in your code below